model/app/fetcher_http.go

```diff
@@ -107,7 +109,25 @@ func (f *httpFetcher) Fetch(src *url.URL, fs appfs.Copier, man Manifest) (err er
 }
 
 func fetchHTTP(src *url.URL, shasum []byte, fs appfs.Copier, man Manifest, prefix string) (err error) {
-	exists, err := fs.Start(man.Slug(), man.Version(), man.Checksum())
+	// Happy path: it exists and we don't need to acquire the lock.
+	exists, err := fs.Exist(man.Slug(), man.Version(), man.Checksum())
+	if err != nil || exists {
+		return err
+	}
+
+	// For the lock key, we use the checksum when available, and else, we
```

Comment:
Why do we use the checksum in the first place? Can we have multiple versions of the same app being installed at the same time?

Reply:
Yes, we can have the same app+version with two checksums being installed/updated at the same time if the app is present in several contexts with different spaces for the apps registry.